src/core/ext/transport/chttp2/transport/flow_control.cc

```diff
@@ -439,75 +287,99 @@ static double get_target_under_memory_pressure(
   return target;
 }
 
-grpc_chttp2_flowctl_action grpc_chttp2_flowctl_get_action(
-    grpc_exec_ctx* exec_ctx, grpc_chttp2_transport_flowctl* tfc,
-    grpc_chttp2_stream_flowctl* sfc) {
-  grpc_chttp2_flowctl_action action;
-  memset(&action, 0, sizeof(action));
+double TransportFlowControl::SmoothLogBdp(grpc_exec_ctx* exec_ctx,
+                                          double value) {
+  grpc_millis now = grpc_exec_ctx_now(exec_ctx);
+  if (!pid_controller_initialized_) {
```

Comment:
Could we just do this in the TransportFlowControl Init() and avoid the is_initialized bool?

Reply:
We'd actually need to back up a little... the initialization depends on 'value', which depends on current memory pressure.

Adding the bool means we can skip duplicating logic, and I'm not clear on how to re-flow this code to avoid that.

Reply:
I'm going to try something though.

Reply:
Got it... LMKWYT

Reply:
Nice, I think this is cleaner